src/compiler/checker.ts

```diff
@@ -5501,11 +5502,11 @@ namespace ts {
                     }
                 }
 
-                function visitAndTransformType<T extends TypeNode>(type: Type, transform: (type: Type) => T) {
+                function visitAndTransformType<T extends Type>(type: T, transform: (type: T) => TypeNode) {

```

Comment:
What was this signature even doing

(rhetorical question, don’t think I want to know)

Reply:
Before it was just plain wrong, that's for sure.